docusaurus/docs/cms/admin-panel-customization/homepage.md

```diff
@@ -0,0 +1,761 @@
+| `label`  | `MessageDescriptor` | The text to display for the link               | Yes      |
+| `href`   | `string`            | The URL where the link should navigate to      | Yes      |
+
+### Creating a widget component
+
+Widget components should be designed to display content in a compact and informative way. 
+
+Here's how to implement a basic widget component:
+
+<Tabs groupId="js-ts">
+<TabItem value="javascript" label="JavaScript">
+
+```jsx title="src/plugins/my-plugin/admin/src/components/MyWidget/index.js"
+import React, { useState, useEffect } from 'react';
+import { Widget } from '@strapi/admin/strapi-admin';
+
+const MyWidget = () => {
+  const [loading, setLoading] = useState(true);
+  const [data, setData] = useState(null);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    // Fetch your data here
+    const fetchData = async () => {
+      try {
+        // Replace with your actual API call
+        const response = await fetch('/my-plugin/data');
+        const result = await response.json();
+        
+        setData(result);
+        setLoading(false);
+      } catch (err) {
+        setError(err);
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  if (loading) {
+    return <Widget.Loading />;
+  }
+
+  if (error) {
+    return <Widget.Error />;
+  }
+
+  if (!data || data.length === 0) {
+    return <Widget.NoData>No data available</Widget.NoData>;
+  }
+
+  return (
+    <div>
+      {/* Your widget content here */}
+      <ul>
+        {data.map((item) => (
+          <li key={item.id}>{item.name}</li>
+        ))}
+      </ul>
+    </div>
+  );
+};
+
+export default MyWidget;
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+```tsx title="src/plugins/my-plugin/admin/src/components/MyWidget/index.tsx"
+import React, { useState, useEffect } from 'react';
+import { Widget } from '@strapi/admin/strapi-admin';
+
+interface DataItem {
+  id: number;
+  name: string;
+}
+
+const MyWidget: React.FC = () => {
+  const [loading, setLoading] = useState<boolean>(true);
+  const [data, setData] = useState<DataItem[] | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    // Fetch your data here
+    const fetchData = async () => {
+      try {
+        // Replace with your actual API call
+        const response = await fetch('/my-plugin/data');
+        const result = await response.json();
+        
+        setData(result);
+        setLoading(false);
+      } catch (err) {
+        setError(err instanceof Error ? err : new Error(String(err)));
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  if (loading) {
+    return <Widget.Loading />;
+  }
+
+  if (error) {
+    return <Widget.Error />;
+  }
+
+  if (!data || data.length === 0) {
+    return <Widget.NoData>No data available</Widget.NoData>;
```

Comment:
FYI if you don't provide children here you get "no content found." as the message. I don't know if you're overriding it intentionally

Reply:
Same as above comment, could you please directly suggest which code to use?